### PR TITLE
Improve the coverage of the README (#995)

### PR DESCRIPTION
**Pull request by KasparEtter**
Preview available at https://github.com/KasparEtter/ts-node/tree/readme

### CODE REVIEW

**Comment by cspotcode** [comment]
Thanks, I made a bunch of tweaks, reverted unrelated formatting changes.

Re the behavior of `--script-mode` when the script is a symlink, we should figure out if that behavior is intentional or a bug.  Better to iron that out first, than to potentially state that a bug is intended via our README.

File: README.md [hunk 1]
```diff
@@ -161,11 +192,27 @@ _Environment variable denoted in parentheses._
...
+* `transformers` `_ts.CustomTransformers | ((p: _ts.Program) => _ts.CustomTransformers)`: An object with transformers or a function that accepts a program and returns an transformers object to pass to TypeScript. Function isn't available with `transpileOnly` flag
+* `readFile`: Custom TypeScript-compatible file reading function
+* `fileExists`: Custom TypeScript-compatible file existence function
+
+### Options via tsconfig.json
+
+Most options can be specified by a `"ts-node"` object in `tsconfig.json` using their programmatic, camelCase names. For example, to enable `--transpile-only`:
+
+```json
+// tsconfig.json
```
Comment (KasparEtter) [inline README.md:206]: Strictly speaking, comments are not allowed in JSON. Given the context is given in the line above, we could/should skip this here.

File: README.md [hunk 2]
```diff
@@ -161,11 +192,27 @@ _Environment variable denoted in parentheses._
...
 * `--log-error` Logs TypeScript errors to stderr instead of throwing exceptions (`TS_NODE_LOG_ERROR`, default: `false`)
 
-### Programmatic Only Options
+### Programmatic-only Options
+
+* `transformers` `_ts.CustomTransformers | ((p: _ts.Program) => _ts.CustomTransformers)`: An object with transformers or a function that accepts a program and returns an transformers object to pass to TypeScript. Function isn't available with `transpileOnly` flag
+* `readFile`: Custom TypeScript-compatible file reading function
+* `fileExists`: Custom TypeScript-compatible file existence function
+
+### Options via tsconfig.json
```
Comment (KasparEtter) [inline README.md:201]: I intentionally had this without a title in the `CLI and Programmatic Options` section as my understanding is that this only applies to those options (but not the CLI-only or programmatic-only options). Now we are mixing with the titles and sections *what* options are available and *how* to provide them. But given that neither approach is perfect, just do you what you think is best.

File: README.md [hunk 3]
```diff
@@ -122,36 +147,42 @@ Create a new Node.js configuration and add `-r ts-node/register` to "Node parame
...
 ### CLI Options
 
-Supports `--print`, `--eval`, `--require` and `--interactive` similar to the [node.js CLI options](https://nodejs.org/api/cli.html).
+`ts-node` supports `--print` (`-p`), `--eval` (`-e`), `--require` (`-r`) and `--interactive` (`-i`) similar to the [node.js CLI options](https://nodejs.org/api/cli.html).
 
-* `--help` Prints help text
-* `--version` Prints version information
+* `-h, --help` Prints the help text
+* `-v, --version` Prints the version. `-vv` prints node and typescript compiler versions, too
+* `-s, --script-mode` Resolve config relative to the directory of the passed script instead of the current directory. Changes default of `--dir`
```
Comment (KasparEtter) [inline README.md:172]: `Resolve` -> `Resolves` for consistency

**Comment by KasparEtter** [comment]
> Re the behavior of --script-mode when the script is a symlink, we should figure out if that behavior is intentional or a bug.

I leave it to you whether you want to create a new issue for this or not. Progress on this would be appreciated, though. 🤓

**Pull request merged**